### PR TITLE
Action Script to automatically run aggregate_data.py on push

### PR DESCRIPTION
**Pull request by RBorn02**

### CODE REVIEW

**Comment by RBorn02** [comment]
I changed the aggregate_data action so that it only runs if a pull request is merged

File: src/export_sql.py [hunk 1]
```diff
@@ -5,28 +5,28 @@
 
 
 def export_to_sql():
-    connection = db.connect('../data/04_sql/database')
+    connection = db.connect('./data/04_sql/database')
 
-    cross_tab = pd.read_parquet('../data/03_experiment_output_aggregated/cross_tab.parquet')
+    cross_tab = pd.read_parquet('./data/03_experiment_output_aggregated/cross_tab.parquet')
```
Comment (Karim-53) [inline src/export_sql.py:10]: for me this relative path doesn't work
`FileNotFoundError: [Errno 2] No such file or directory: './data/03_experiment_output_aggregated/cross_tab.parquet'`

Please use a root variable

```python
from src.utils import root

def export_to_sql():
    connection = db.connect(root + '/data/04_sql/database')
```

File: .github/workflows/aggregate_data.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        with:
+          python-version: 3.8
+      - name: install python dependencies
+        uses: py-actions/py-dependency-install@v4
+        with:
+          path: "requirements.txt"
+      - name: run aggregate_data.py
+        shell: bash
+        run: |
+            pip install pyarrow
```
Comment (Karim-53) [inline .github/workflows/aggregate_data.yml:28]: would you prefer to add pyarrow directly to `requirements.txt` ?

File: .github/workflows/aggregate_data.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
+name: Run aggregate_data.py
+
+on:
+  pull_request:
+    types:
+      - closed
+
+jobs:
+  build:
+    if: github.event.pull_request.merged == true
```
Comment (Karim-53) [inline .github/workflows/aggregate_data.yml:10]: we need to know that this script works perfectly before merging the PR

File: .github/workflows/aggregate_data.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout repo content
+        uses: actions/checkout@v3 # checkout the repository content to github runner.
```
Comment (Karim-53) [inline .github/workflows/aggregate_data.yml:14]: use v3 or v4 to skip the warning that will become an error this summer
image

File: explainers/joint_shapley.py [hunk 5]
```diff
@@ -1,10 +1,9 @@
-import torch
 import shap
 import numpy as np
 import pandas as pd 
 import random
 import sys
-from IPython.display import display, clear_output
+#from IPython.display import display, clear_output
```
Comment (Karim-53) [inline explainers/joint_shapley.py:6]:
```suggestion
```

**Review by Karim-53 — APPROVED** [review]

**Pull request merged**